pandas/tests/groupby/test_transform.py

```diff
@@ -880,3 +880,13 @@ def test_transform_absent_categories(func):
     result = getattr(df.y.groupby(df.x), func)()
     expected = df.y
     assert_series_equal(result, expected)
+
+
+def test_transform_empty_df():
```

Comment:
make the test name: test_evalute_with_empty_groups.

move to test_grouping

parameterize using .transform, .apply, and .agg (should be the same results) (you can pass in the function name, then use ``getattr`` inside)

Reply:
Am having trouble implementing this, because I get an error on index mismatch when func is 'apply' or 'agg'.  Transform uses self._selected_obj.index, which is a RangeIndex, whereas apply/agg calls self._python_apply/agg_general, which returns a Series with Float64Index.

{AssertionError}Series.index are different

Series.index classes are not equivalent
[left]:  Float64Index([], dtype='float64', name=1)
[right]: RangeIndex(start=0, stop=0, step=1)

Reply:
The various functions would create empty series with different index objects. This seems to be a bigger issue than what I am addressing. (Perhaps they all should use RangeIndex?) 

Given this, can I leave the test in test_transform.py?

Reply:
no, pls move the test as indicated